### ISSUE TITLE
Redstone state changes broken if GP installed and eigen redstone enabled

### ISSUE THREAD

**Issue by i9hdkill** [SpongePowered/SpongeForge#2796]
**I am currently running**

- SpongeForge version: 3755
- Forge version: 2809

- Plugins/Mods: Minecraft, Minecraft Coder Pack, SpongeAPI, Sponge, GriefPrevention, LuckPerms

**Issue Description**
Redstone updates seems to be broken if GP installed and eigen redstone enabled. 
How to reproduce:
Setup a fresh world with just SF and GP installed and eigen redstone *disabled* and build this circuit: (No claims or config settings required)
![download_pre](https://user-images.githubusercontent.com/4679374/58811071-677f1280-861f-11e9-8c1c-31b1a349a9fe.jpg)
Now remove the redstone dust at the cursor poisition. It should do that:
![Screenshot_3](https://user-images.githubusercontent.com/4679374/58811148-90070c80-861f-11e9-943c-ef80d8532014.jpg)

Now stop your server, enable eigen redstone and test the same again.
It produces that:
![download](https://user-images.githubusercontent.com/4679374/58811419-13286280-8620-11e9-9a49-3d439a66918f.jpg)

Conclusion:
SF + eigen redstone == fine
SF + Normal redstone == fine
SF + GP + Normal redstone == fine
SF + GP + eigen redstone == broken

No errors or similiar in log

**Comment by gabizou** [comment]
And just to confirm: this also reproduces if `-Dsponge.shouldFireAll=true` without GP? @bloodmc would you know off the top of your head what sort of listener context might be happening here?

**Comment by johnfriedrich** [comment]
Yep, happens with this flag even without GP. That's the most basic setup I could find, but as you might think, this breaks everything, even the most simple door mechanisms

**Comment by gabizou** [comment]
So the problem is with eigen redstone and tracking enabled.
